lib/matplotlib/_mathtext.py

```diff
@@ -27,6 +27,11 @@
 from .font_manager import FontProperties, findfont, get_font
 from .ft2font import FT2Image, KERNING_DEFAULT
 
+from pyparsing import __version__ as pyparsing_version
+if pyparsing_version < '3.0.0':
```

Comment:
This falls into the category of lurking problems that _probably_ won't be a real problem but _could_ be...

string comparison of version numbers is likely to fail. 

Unfortunately the version we are gaurding against doesn't have a tuple version (with integers) to compare to. (Current pyparsing _does_)

we are saved a bit by this having minor/patch version of 0, but would still fail if pyparsing got to version `10.0.0` as we would want that to be greater than 3.0.0, but string comparison would fail. Now _probably_ by the time that happens (_if_ it happens) we will no longer support versions less than 3... but if pyparsing decided to switch to date versions of `2023.7.0` or something, we would be in the same boat a whole lot sooner (and possibly without warning).

```suggestion
if tuple(int(x) for x in pyparsing_version.split(".")) < (3, 0, 0):
```

may work in a bit more robust manner

Or we can just use `nestedExpr` for now and handle switching to the new name later (as at least oneOf is already using the deprecated version, so that change can be separated from this PR.